contracts/examples/SimpleInterestTermsContract.sol

```diff
@@ -118,15 +137,44 @@ contract SimpleInterestTermsContract {
     function unpackParameters(bytes32 parameters)
         public
         pure
-        returns (uint128 _principalPlusInterest, uint128 _termLengthInBlocks)
+        returns (
+            uint128 _principalPlusInterest,
+            uint8 _amortizationUnitType,
+            uint120 _termLengthInAmortizationUnits
+        )
     {
-        bytes16[2] memory values = [bytes16(0), 0];
+        bytes16 principalPlusInterest;
```

Comment:
I think when we go down into assembly we should be super intentional about documenting why.